.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,77 @@
+name: unit-tests
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - 'server/**'
+      - 'tests/functional/**'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - 'server/**'
+      - 'tests/functional/**'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  unit-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: linux
+            runs-on: ubuntu-22.04
+            arch: x86_64
+
+    name: "${{ matrix.os }}-${{ matrix.arch }}"
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Install Nim
+        uses: iffy/install-nim@0ed218de08059d18507144e6aee70e6bad0bc148 # v5.0.3
+        with:
+          version: "binary:2.0.0"
+        env:
+          GITHUB_TOKEN: ${{ github.token }} # Avoid rate limiting.
+
+      - name: Install Nimble dependencies
+        run: nimble --accept install --depsOnly
+
+      # for tests to checkout dummy private repo
+      - name: Get GitHub Token
+        id: org-token
+        uses: crashappsec/action-github-app-token@main
+        with:
+          app_id: ${{ secrets.CHALK_GITHUB_APP_ID }}
+          private_key: ${{ secrets.CHALK_GITHUB_APP_PRIVATE_KEY }}
+
```

Comment:
this is only used in integration tests to checkout dummy repos as noted in the comment. we should def not need it for unit tests